modelator/utils/apalache_helpers.py

```diff
@@ -15,14 +15,21 @@ def extract_tla_module_name(tla_file_content: str):
     return match.group("moduleName")
 
 
-def write_trace_files_to(apalache_result: Dict, traces_dir: str) -> List[str]:
+def write_trace_files_to(
+    apalache_result: Dict, traces_dir: str, simulate: bool = False
+) -> List[str]:
     # create directory if it does not exist
     Path(traces_dir).mkdir(parents=True, exist_ok=True)
 
+    itfs_filenames_pattern = re.compile(r"\d\.itf\.json$")
     itfs_filenames = [
-        f for f in apalache_result["files"].keys() if f.endswith(".itf.json")
+        f for f in apalache_result["files"].keys() if itfs_filenames_pattern.search(f)
     ]
     itfs_filenames.sort()
+    if simulate is True and len(itfs_filenames) > 0:
+        # had to add [1:] because apalache simulate produces one extra trace (and I suspect
+        # that example0.itf.json and example1.itf.json are the same)
+        itfs_filenames = itfs_filenames[1:]
```

Comment:
This will break if  Apalache stops producing `example0.itf.json`. Can we do a concrete filter here?
```
itfs_zero_pattern = re.compile(r"[^\d]0\.itf\.json$")
itfs_filenames = [x for x in itfs_filenames if itfs_zero_pattern.search(x)]
```

Reply:
Cool. Btw I just noticed my filter is wrong, it should be not of the regex success.